apps/publisher/controllers/apis-router.jag

```diff
@@ -35,44 +35,47 @@ var mapper = function(path) {
         return path;
     };
 };
-metrics.start("apis-router","router");
-if (uriMatcher.match(API_URL)) {
-    var args = uriMatcher.elements();
-    var tenantId = DEFAULT_TENANT; //Assume that the tenant will be the super tenant
-    //Determine the tenant by the user
-    if (user) {
-        tenantId = user.tenantId;
-    }
-    //Obtain the set of available pages
-    var app = require('rxt').app;
-    var page = getPage(args.suffix);
-    var path = app.getApiEndpointPath(tenantId, page);
-    if (path) {
+try{
+    metrics.start("apis-router","router");
+    if (uriMatcher.match(API_URL)) {
+        var args = uriMatcher.elements();
+        var tenantId = DEFAULT_TENANT; //Assume that the tenant will be the super tenant
+        //Determine the tenant by the user
+        if (user) {
+            tenantId = user.tenantId;
+        }
+        //Obtain the set of available pages
+        var app = require('rxt').app;
+        var page = getPage(args.suffix);
+        var path = app.getApiEndpointPath(tenantId, page);
+        if (path) {
 
-        var file = new File(path);
-        if (file.isExists()) {
-            request.getMappedPath = mapper(path);
-            var username = user ? user.username : null;
-            var isAuthorized = rxtAPI.permissions.hasAppAPIPermission(page,tenantId,username);
-            if(!isAuthorized){
-                log.error('user '+username+' does not have permission to access the api '+page);
-                response.sendError(401,'You do not have access to this api');
-            } else {          
-                var proceed = app.execApiHandlers('onApiLoad', request, response, session, page);
-                if (proceed) {
-                    include(path);
+            var file = new File(path);
+            if (file.isExists()) {
+                request.getMappedPath = mapper(path);
+                var username = user ? user.username : null;
+                var isAuthorized = rxtAPI.permissions.hasAppAPIPermission(page,tenantId,username);
+                if(!isAuthorized){
```

Comment:
Thanks @mushthaq33 , I have fixed it in [this commit](https://github.com/wso2/carbon-store/pull/844/commits/931b708ba54b43643839e3b48b30a597c4f4e0d8)